Tests/UnitTests/EdgeHitProcessorTests.swift

```diff
@@ -241,18 +241,27 @@ class EdgeHitProcessorTests: XCTestCase {
         expectation.expectedFulfillmentCount = recoverableNetworkErrorCodes.count
         let event = Event(name: "test-event", type: EventType.custom, source: EventSource.requestContent, data: nil)
 
-        for code in recoverableNetworkErrorCodes {
+        // (headerValue, actualRetryValue)
+        let retryValues = [("60", 60.0), ("InvalidHeader", 5.0), ("", 5.0), ("1", 1.0)]
+
+        for (code, retryValueTuple) in zip(recoverableNetworkErrorCodes, retryValues) {
             let error = EdgeEventError(title: "test-title", detail: nil, status: code, type: "test-type", eventIndex: 0, report: nil)
             let edgeResponse = EdgeResponse(requestId: "test-req-id", handle: nil, errors: [error], warnings: nil)
             let responseData = try? JSONEncoder().encode(edgeResponse)
 
-            mockNetworkService?.connectAsyncMockReturnConnection = HttpConnection(data: responseData, response: HTTPURLResponse(url: URL(string: "adobe.com")!, statusCode: code, httpVersion: nil, headerFields: nil), error: nil)
+            mockNetworkService?.connectAsyncMockReturnConnection = HttpConnection(data: responseData,
+                                                                                  response: HTTPURLResponse(url: URL(string: "adobe.com")!,
+                                                                                  statusCode: code,
+                                                                                  httpVersion: nil,
+                                                                                  headerFields: ["Retry-After": retryValueTuple.0]),
```

Comment:
Is this header read by the network service as int from the server response and stored as string in the HttpConnection or do we need to add support for other header types?

Reply:
HTTP headers are string/string key values, I don't believe there is any need to support other values. If you look at other networking libraries you can see that only [String: String] headers are supported.

https://stackoverflow.com/questions/34152142/possible-types-of-a-http-header-value#:~:text=The%20type%20of%20a%20value,string%20is%20actually%20a%20string.

Reply:
It's strange to me they define it as `[AnyHashable: Any]` because they only allow you to read and index them as strings.
https://developer.apple.com/documentation/foundation/httpurlresponse/3240613-value